guvnor-ala/guvnor-ala-docker-provider/src/main/java/org/guvnor/ala/docker/access/impl/DockerAccessInterfaceImpl.java

```diff
@@ -0,0 +1,63 @@
+
+package org.guvnor.ala.docker.access.impl;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.spotify.docker.client.DefaultDockerClient;
+import com.spotify.docker.client.DockerCertificateException;
+import com.spotify.docker.client.DockerClient;
+import com.spotify.docker.client.DockerException;
+import com.spotify.docker.client.messages.Info;
+import org.uberfire.commons.lifecycle.Disposable;
+import org.guvnor.ala.docker.access.DockerAccessInterface;
+import org.guvnor.ala.runtime.providers.ProviderId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DockerAccessInterfaceImpl
+        implements DockerAccessInterface,
+        Disposable {
+
+    protected static final Logger LOG = LoggerFactory.getLogger( DockerAccessInterfaceImpl.class );
+    final Map<String, DockerClient> clientMap = new HashMap<>();
+
+    @Override
+    public DockerClient getDockerClient( final ProviderId providerId ) throws DockerException, InterruptedException {
```

Comment:
It appears that ```buildClient(...)``` runs asynchronously (due to use of ```InterruptedException```). If this is the case should this method not be synchronised (or at least use of the ```clientMap```)?